tests/src/e2e.rs

```diff
@@ -406,11 +411,246 @@ mod tests {
+            ALBERT,
+            "--min-amount-buy",
+            "50",
+            "--max-amount-sell",
+            "300",
+            "--token-buy",
+            BTC,
+            "--token-sell",
+            ETH,
+            "--min-rate",
+            "0.7",
+            "--file-path",
+            intent_b_path.as_ref(),
+        ];
+        let mut craft_intent_b = Command::cargo_bin("anomac")?;
+        craft_intent_b.args(tx_b);
+        craft_intent_b.output().expect("Should create the intent");
+
+        // cargo run --bin anomac -- craft-intent --key $CHRISTEL --address
+        // $CHRISTEL --min-amount-buy 20 --max-amount-sell 200 --token-buy $ETH
+        // --token-sell $XAN --min-rate 0.5 --file-path intent.C
+        let tx_c = vec![
+            "craft-intent",
+            "--key",
+            CHRISTEL,
+            "--address",
+            CHRISTEL,
+            "--min-amount-buy",
+            "20",
+            "--max-amount-sell",
+            "320000",
+            "--token-buy",
+            ETH,
+            "--token-sell",
+            XAN,
+            "--min-rate",
+            "0.5",
+            "--file-path",
+            intent_c_path.as_ref(),
+        ];
+        let mut craft_intent_c = Command::cargo_bin("anomac")?;
+        craft_intent_c.args(tx_c);
+        craft_intent_c.output().expect("Should create the intent");
+
+        //  Start gossip
+        let mut session_gossip = spawn_command(base_node_gossip, Some(40_000))
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        //  Start ledger
+        let mut session_ledger = spawn_command(base_node_ledger, Some(40_000))
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        session_ledger
+            .exp_string(&"No state could be found".to_string())
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // Wait for ledger and gossip to start
+        sleep(3);
+
+        // cargo run --bin anomac -- intent --node "http://127.0.0.1:39111" --data-path intent.A --topic "asset_v1"
+        // cargo run --bin anomac -- intent --node "http://127.0.0.1:39112" --data-path intent.B --topic "asset_v1"
+        // cargo run --bin anomac -- intent --node "http://127.0.0.1:39112" --data-path intent.C --topic "asset_v1"
+        //  Send intent A
+        let mut send_intent_a = Command::cargo_bin("anomac")?;
+        send_intent_a.args(&[
+            "intent",
+            "--node",
+            "http://127.0.0.1:39111",
+            "--data-path",
+            intent_a_path.as_ref(),
+            "--topic",
+            "asset_v1",
+        ]);
+
+        let mut session_send_intent_a =
+            spawn_command(send_intent_a, Some(20_000))
+                .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // means it sent it correctly but not able to gossip it (which is
+        // correct since there is only 1 node)
+        session_send_intent_a
+            .exp_regex(".*Failed to publish_intent InsufficientPeers*")
```

Comment:
I think when there's no regex, you can just use
```suggestion
            .exp_string("Failed to publish_intent InsufficientPeers")
```
same for all other `exp_regex` below